codes/c/chapter_greedy/max_capacity.c

```diff
@@ -0,0 +1,37 @@
+// File: max_capacity.c
+// Created Time: 2023-09-15
+// Author: xianii (xianyi.xia@outlook.com)
+
+#include "../utils/common.h"
+
+/* 最大容量：贪心 */
+int maxCapacity(int ht[], int htLength) {
+	// 初始化 i, j 分列数组两端
+	int i = 0;
+	int j = htLength - 1;
+	// 初始最大容量为 0
+	int res = 0;
+	// 循环贪心选择，直至两板相遇
+	while(i < j) {
+		// 更新最大容量
+		int capacity = (ht[i] < ht[j] ? ht[i] : ht[j]) * (j - i);
+		res = res > capacity ? res : capacity;
```

Comment:
It is recommended to define two macros `MIN` and `MAX` for better understanding.